src/d2r_image/ocr.py

```diff
@@ -0,0 +1,255 @@
+    # case: two 1's within a string; e.g., "S11PER MANA POTION"
+    try:
+        text = ONEONE_U.sub('U', text)
+    except:
+        # logging.error(f"Error _11_ -> _U_ on {ocr_output}")
+        pass
+    # case: an I within a number or by a sign; e.g., "+32I to mana attack rating"
+    try:
+        text = I_1.sub('1', text)
+    except:
+        # logging.error(f"Error I -> 1 on {ocr_output}")
+        pass
+    # case: a 1 within a string; e.g., "W1RT'S LEG"
+    try:
+        text = ONE_I.sub('I', text)
+    except:
+        # logging.error(f"Error 1 -> I on {ocr_output}")
+        pass
+    # case: a solitary I; e.g., " I TO 5 DEFENSE"
+    cnt = 0
+    while True:
+        cnt += 1
+        if cnt > 30:
+            # logging.error(f"Error ' I ' -> ' 1 ' on {ocr_output}")
+            break
+        if " I " in text:
+            text = text.replace(" I ", " 1 ")
+            continue
+        elif ' I\n' in text:
+            text = text.replace(' I\n', ' 1\n')
+            continue
+        elif '\nI ' in text:
+            text = text.replace('\nI ', '\n1 ')
+            continue
+        break
+    # case: a solitary S; e.g., " 1 TO S DEFENSE"
+    cnt = 0
+    while True:
+        cnt += 1
+        if cnt > 30:
+            # logging.error(f"Error ' S ' -> ' 5 ' on {ocr_output}")
+            break
+        if " S " in text:
+            text = text.replace(" S ", " 5 ")
+            continue
+        elif ' I\n' in text:
+            text = text.replace(' S\n', ' 5\n')
+            continue
+        elif '\nI ' in text:
+            text = text.replace('\nS ', '\n5 ')
+            continue
+        break
+    # case: consecutive I's; e.g., "DEFENSE: II"
+    repeat = False
+    cnt = 0
+    while "II" in text:
+        cnt += 1
+        if cnt > 30:
+            # logging.error(f"Error 4 on {ocr_output}")
+            break
+        text = text.replace("II", "11")
+        repeat = True
+        repeat_count += 1
+    if repeat and repeat_count < 10:
+        _fix_regexps(text)
+    return text
+
+
+def _check_known_errors(text):
+    for key in ERROR_RESOLUTION_MAP:
+        if key in text:
+            text = text.replace(key, ERROR_RESOLUTION_MAP[key])
+    return text
+
+def _ocr_result_dictionary_check(
+    original_text: str,
+    confidences: list,
+    word_list: list = all_words(),
+    normalized_lev_threshold: float = 0.6,
+    ocr_confidence_threshold: float = 0.9
+    ) -> str:
+    confidences = [x/100 for x in confidences]
+    if all(x >= ocr_confidence_threshold for x in confidences):
+        return original_text
+    #print(f"{original_text} {confidences}")
+    text = original_text.replace('\n', ' NEWLINEHERE ')
+    corrected_text = ""
+    word_count = 0
+    for word in text.split(' '):
+        word = word.strip()
+        if word and word != "NEWLINEHERE":
+            try:
+                #print(confidences[word_count])
+                if confidences[word_count] <= ocr_confidence_threshold:
```

Comment:
It might actually be good to filter on whether or not the word is already *in* the word list before trying to fuzzy find it.  In fact, it might be worth going to the fuzzy finder even with a high ocr confidence, if the word isn't one that we recognize.

Reply:
Part of this is that I'm confident I have like... 99.9% of the strings in the wordlist. If OCR has a high confidence on a word then I suspect it's probably a true word. Had to add a bunch of words to the wordlist tonight because I've chosen now to correct words that have punctuation attached like `Cost:` `Value:`. By not checking every word i give a little wiggle room for cases I might not have anticipated.